chips/msp432/src/i2c.rs

```diff
@@ -0,0 +1,374 @@
+        }
+    }
+
+    fn enable_interrupts(&self) {
+        // Enable interrupts
+        self.registers.ie.modify(
+            // Enable NACK interrupt
+            usci::UCBxIE::UCNACKIE::SET
+            // Enable TX interrupt
+            // + usci::UCBxIE::UCTXIE0::SET
+            // Enable RX interrupt
+            + usci::UCBxIE::UCRXIE0::SET
+            // Enable Stop condition interrupt
+            + usci::UCBxIE::UCSTPIE::SET
+            // Enable Start condition interrupt
+            + usci::UCBxIE::UCSTTIE::SET
+            // Enable 'arbitration lost' interrupt
+            + usci::UCBxIE::UCALIE::SET,
+        );
+    }
+
+    fn enable_transmit_mode(&self) {
+        self.registers
+            .ctlw0
+            .modify(usci::UCBxCTLW0::UCTR::Transmitter);
+    }
+
+    fn enable_receive_mode(&self) {
+        self.registers.ctlw0.modify(usci::UCBxCTLW0::UCTR::Receiver);
+    }
+
+    fn enable_transmit_interrupt(&self) {
+        self.registers.ie.modify(usci::UCBxIE::UCTXIE0::SET);
+    }
+
+    fn disable_transmit_interrupt(&self) {
+        self.registers.ie.modify(usci::UCBxIE::UCTXIE0::CLEAR);
+    }
+
+    fn set_byte_counter(&self, val: u8) {
+        self.registers.tbcnt.set(val as u16);
+    }
+
+    fn invoke_callback(&self, err: i2c::Error) {
+        // Reset buffer index and set mode to Idle in order to start a new transfer properly
+        self.buf_idx.set(0);
+        self.mode.set(OperatingMode::Idle);
+
+        self.buffer.take().map(|buf| {
+            self.master_client
+                .map(move |cl| cl.command_complete(buf, err))
+        });
+    }
+
+    fn setup(&self) {
+        self.set_module_to_reset();
+
+        self.registers.ctlw0.modify(
+            // Use 7 bit addresses
+            usci::UCBxCTLW0::UCSLA10::AddressSlaveWith7BitAddress
+            // Setup to master mode
+            + usci::UCBxCTLW0::UCMST::MasterMode
+            // Setup to single master environment
+            + usci::UCBxCTLW0::UCMM::SingleMasterEnvironment
+            // Configure USCI module to I2C mode
+            + usci::UCBxCTLW0::UCMODE::I2CMode
+            // Enable Synchronous mode
+            + usci::UCBxCTLW0::UCSYNC::SynchronousMode
+            // Set clock source to SMCLK (1.5MHz)
+            + usci::UCBxCTLW0::UCSSEL::SMCLK,
+        );
+
+        self.registers.ctlw1.modify(
+            // Disable clock low timeout
+            usci::UCBxCTLW1::UCCLTO::CLEAR
+            // Send a NACK before a stop condition
+            + usci::UCBxCTLW1::UCSTPNACK::NackBeforeStop
+            // Generate the ACK bit by hardware
+            + usci::UCBxCTLW1::UCSWACK::HardwareTriggered
+            // Set glitch filtering to 50ns (according to I2C standard)
+            + usci::UCBxCTLW1::UCGLIT::_50ns,
+        );
+
+        // Don't clear the module reset here since we set the state to Disabled
+        self.mode.set(OperatingMode::Disabled);
+    }
+
+    pub fn set_speed(&self, speed: Speed) {
```

Comment:
Does this need to be `pub`?

Reply:
Since the HIL traits don't offer any possibility to set the speed it's done via a seperate function. At least at the STM32F4 discovery board (which is where I got the idea from) it's done the same way.

I could also set it to a default value if this would be a better solution.